pkg/authorization/admission/restrictusers/subjectchecker.go

```diff
@@ -0,0 +1,319 @@
+package restrictusers
+
+import (
+	"fmt"
+
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	oclient "github.com/openshift/origin/pkg/client"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	usercache "github.com/openshift/origin/pkg/user/cache"
+)
+
+// SubjectChecker determines whether rolebindings on a subject (user, group, or
+// service account) are allowed in a project.
+type SubjectChecker interface {
+	Allowed(kapi.ObjectReference, *RoleBindingRestrictionContext) (bool, error)
+}
+
+// UnionSubjectChecker represents the union of zero or more SubjectCheckers.
+type UnionSubjectChecker []SubjectChecker
+
+// NewUnionSubjectChecker returns a new UnionSubjectChecker.
+func NewUnionSubjectChecker(checkers []SubjectChecker) (UnionSubjectChecker, error) {
+	return UnionSubjectChecker(checkers), nil
+}
+
+// Allowed determines whether the given subject is allowed in rolebindings in
+// the project.
+func (checkers UnionSubjectChecker) Allowed(subject kapi.ObjectReference,
+	ctx *RoleBindingRestrictionContext) (bool, error) {
+	for _, checker := range []SubjectChecker(checkers) {
+		allowed, err := checker.Allowed(subject, ctx)
+		if err != nil {
+			return false, err
```

Comment:
Track errors and report them as an aggregate at the end.  I think its reasonable to keep checking other `SubjectCheckers` in the case of error.